lua/gitlab/reviewer/diffview.lua

```diff
@@ -21,7 +28,20 @@ M.open = function()
     return
   end
 
-  vim.api.nvim_command(string.format("DiffviewOpen %s..%s", diff_refs.base_sha, diff_refs.head_sha))
+  local diffview_open_command = "DiffviewOpen"
+
+  if state.settings.reviewer_settings.diffview.imply_local then
+    if are_git_managed_files_modified() then
+      u.notify(
+        "There are uncommited changes in the working tree, cannot use 'imply_local' setting for gitlab reviews. Stash or commit all changes to use.",
+        vim.log.levels.WARN
+      )
+    else
+      diffview_open_command = diffview_open_command .. " --imply-local"
+    end
+  end
+
+  vim.api.nvim_command(string.format("%s %s..%s", diffview_open_command, diff_refs.base_sha, diff_refs.head_sha))
```

Comment:
When using a messenger tool like [notify](https://github.com/rcarriga/nvim-notify), the message is lost when switching tabs. We want to flash the warning after the reviewer changes tabs, so you'd have to store it in a variable:

```lua
  local has_unmodified_files = are_git_managed_files_modified()
  if state.settings.reviewer_settings.diffview.imply_local and not has_unmodified_files then
    diffview_open_command = diffview_open_command .. " --imply-local"
  end

  -- The code to open the reviewer in a new tab...

  if has_unmodified_files and state.settings.reviewer_settings.diffview.imply_local then
    u.notify(
      "There are uncommited changes in the working tree, cannot use 'imply_local' setting for gitlab reviews. Stash or commit all changes to use.",
      vim.log.levels.WARN
    )
  end
```